app/api/probe_dock/results_api.rb

```diff
@@ -0,0 +1,82 @@
+# Copyright (c) 2015 ProbeDock
+# Copyright (c) 2012-2014 Lotaris SA
+#
+# This file is part of ProbeDock.
+#
+# ProbeDock is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ProbeDock is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ProbeDock.  If not, see <http://www.gnu.org/licenses/>.
+module ProbeDock
+  class ResultsApi < Grape::API
+    namespace :results do
+      before do
+        authenticate
+      end
+
+      helpers do
+        def current_organization
+          @current_organization ||= if params[:reportId].present?
+            TestReport.where(api_id: params[:reportId]).first!.organization
+          elsif params[:projectId].present?
+            Project.where(api_id: params[:projectId]).first!.organization
+          elsif params[:projectVersionId].present?
+            ProjectVersion.where(api_id: params[:projectVersionId]).first!.project.organization
+          elsif params[:testId].present?
+            ProjectTest.where(api_id: params[:testId]).first!.project.organization
+          end
+        end
+      end
+
+      get do
+        authorize! TestResult, :index
+
+        rel = TestResult
+          .order('test_results.active desc, test_results.passed, test_results.name, test_results.id')
```

Comment:
This order clause was meant for display to display the results in test reports, where we display failing results first, then passing ones, then inactive ones.

In the case of the test widget, we'll want to order by run date.

I would leave this as the default order, but allow the user to give a `sort=runAt` param which will change the order.